frontend/util.lua

```diff
@@ -1344,4 +1344,73 @@ function util.stringEndsWith(str, ending)
    return ending == "" or str:sub(-#ending) == ending
 end
 
+local WrappedFunction_mt = {
+    __call = function(self, ...)
+        if self.before_callback then
+            self.before_callback(self.target_table)
+        end
+        if self.func then
+            return self.func(...)
+        end
+    end,
+}
+
+--- Wrap (or replace) a table method with a custom method, in a revertable way.
+-- This is primarily useful for interfaces where it is necessary to wrap table
+-- methods as a method of extension (most notably, wrapping inputtext for some
```

Comment:
Not sure I really understand the sentence "to wrap table methods as a method of extension".
(I know it's hard to write such documentation for such abstracts use cases :)

Reply:
I'm basically trying to describe how the inputtext extension system works more generically -- where you modify (usually wrap) the methods of a table in order to augment the default behaviour and extend it in some way.